Framework/Core/src/DataSpecUtils.cxx

```diff
@@ -510,7 +511,8 @@ DataDescriptorMatcher DataSpecUtils::dataDescriptorMatcherFrom(ConcreteDataTypeM
   auto timeDescriptionMatcher = std::make_unique<DataDescriptorMatcher>(
     DataDescriptorMatcher::Op::And,
     DescriptionValueMatcher{dataType.description.as<std::string>()},
-    StartTimeValueMatcher(ContextRef{0}));
+    std::make_unique<DataDescriptorMatcher>(DataDescriptorMatcher::Op::Just,
```

Comment:
Why would you add the extra node?

Reply:
yes, understood but then should't we support both cases? the matcher with three ands is a perfectly valid one.

Reply:
yes, understood but then should't we support both cases? the matcher with three ands is a perfectly valid one.

Reply:
ok, my change was more in terms of code, not functionality. Its not required and not part of the actual fix in this PR. It only adopts the code structure of the other methods.